File: programmers/unanchoi/lv2/N개의최소공배수.kt
class Solution {
    fun gcd(a: Int, b: Int) : Int {
        return if (b==0) a else gcd(b, a % b)
    }

    fun lcm(a: Int, b: Int) : Int {
        return a * b / gcd(a, b)
    }

    fun solution(arr: IntArray): Int {
        var answer = arr[0]
        for (i in 1 until arr.size) {
            answer = lcm(answer, arr[i])
        }
        return answer
    }
}

File: programmers/algorithm/level2/Hong0329/N개의 최소공배수.cpp
//
// Created by hong jun chung on 6/6/24.
//
#include <string>
#include <vector>
#include <algorithm>
using namespace std;
 
int fun(int a , int b)
{
    int A = max(a, b);
    int B = min(a, b);
    
    while(A % B != 0)
    {
        int R = A % B;
        A = B;
        B = R;
    }
    return B;
}
 

int solution(vector<int> arr) 
{
    int answer = arr[0];
    for(int i = 1; i < arr.size(); i++)
    {
        int x = fun(answer, arr[i]);
        int y = answer * arr[i] / x;
        answer = y;
    }
    return answer;
}